pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -711,23 +686,169 @@ func (r *ReconcileChaosEngine) addFinalzerToEngine(engine *chaosTypes.EngineInfo
 		deletetimeStamp != nil {
 		return true
 	}
 	return false
 }
+
+func (r *ReconcileChaosEngine) checkRunnerPodCompleted(engine *chaosTypes.EngineInfo) bool {
+	runnerPod := corev1.Pod{}
+	r.client.Get(context.TODO(), types.NamespacedName{Name: engine.Instance.Name + "-runner", Namespace: engine.Instance.Namespace}, &runnerPod)
+	return runnerPod.Status.Phase == corev1.PodSucceeded
+}
+
+func (r *ReconcileChaosEngine) removeDefaultChaosResources(request reconcile.Request) (reconcile.Result, error) {
+	if err := r.removeChaosRunner(engine, request); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := r.removeChaosServices(engine, request); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileChaosEngine) removeChaosRunner(engine *chaosTypes.EngineInfo, request reconcile.Request) error {
+	optsList := []client.ListOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"app": engine.Instance.Name, "chaosUID": string(engine.Instance.UID)},
+	}
+	var podList corev1.PodList
+	if errList := r.client.List(context.TODO(), &podList, optsList...); errList != nil {
+		return errList
+	}
+	for _, v := range podList.Items {
+		if errDel := r.client.Delete(context.TODO(), &v, []client.DeleteOption{}...); errDel != nil {
+			return errDel
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileChaosEngine) checkRunnerPodForCompletion(engine *chaosTypes.EngineInfo, reqLogger logr.Logger) error {
+	isCompleted := r.checkRunnerPodCompleted(engine)
+	if isCompleted {
+		err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusCompleted)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkEngineStatusForComplete(engine *chaosTypes.EngineInfo) bool {
+	return engine.Instance.Status.EngineStatus == litmuschaosv1alpha1.EngineStatusCompleted
+
+}
+func (r *ReconcileChaosEngine) reconcileForComplete(request reconcile.Request) (reconcile.Result, error) {
+	_, err := r.removeDefaultChaosResources(request)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if engine.Instance.ObjectMeta.Finalizers != nil {
+		engine.Instance.ObjectMeta.Finalizers = utils.RemoveString(engine.Instance.ObjectMeta.Finalizers, "chaosengine.litmuschaos.io/finalizer")
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "ChaosEngine Stopped", "Removing all experiment resources allocated to ChaosEngine: %v in Namespace: %v", engine.Instance.Name, engine.Instance.Namespace)
+	}
+	err = r.updateState(engine, litmuschaosv1alpha1.EngineStateStop)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileChaosEngine) initEngineState(engine *chaosTypes.EngineInfo) error {
+	if engine.Instance.Spec.EngineState == "" {
+		err := r.updateState(engine, litmuschaosv1alpha1.EngineStateActive)
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileChaosEngine) initEngineStatus(engine *chaosTypes.EngineInfo) error {
```

Comment:
Can we do this function operation inside `initEngineState` function, like both are doing the operation at initializing time